x-pack/solutions/observability/plugins/apm/server/routes/alerts/rule_types/error_count/register_error_count_rule_type.ts

```diff
@@ -278,7 +283,13 @@ export function registerErrorCountRuleType({
         const recoveredAlertId = recoveredAlert.alert.getId();
         const alertUuid = recoveredAlert.alert.getUuid();
         const alertDetailsUrl = getAlertDetailsUrl(basePath, spaceId, alertUuid);
-        const groupByFields: Record<string, string> = allGroupByFields.reduce(
+
+        const ruleParamsOfRecoveredAlert = alertHits?.[
+          ALERT_RULE_PARAMETERS
+        ] as ErrorCountRuleTypeParams;
+        const groupByFieldsOfRecoveredAlert = ruleParamsOfRecoveredAlert.groupBy ?? [];
+
+        const groupByFields: Record<string, string> = groupByFieldsOfRecoveredAlert.reduce(
```

Comment:
Q: What is the difference between `allGroupByFields` (which was previously used) and `groupByFieldsOfRecoveredAlert`? (We discussed this briefly, but maybe we can log the discussion here for others as well)

Reply:
as it turns out, we need to use `getAllGroupByFields` after all, as when rule is saved without changing group by fields, `rule.params.groupBy` is empty so we need to manually get default group by fields.

Addressed in 4503f0dc8931dbb22ad99d3c941f94a7ec722891